workflows/utilities/wf_dorado_basecalling.wdl

```diff
@@ -0,0 +1,78 @@
+version 1.0
+
+import "../../tasks/basecalling/task_dorado_basecall.wdl" as basecall_task
+import "../../tasks/basecalling/task_samtools_convert.wdl" as samtools_convert_task
+import "../../tasks/basecalling/task_dorado_demux.wdl" as dorado_demux_task
+import "../../tasks/utilities/file_handling/task_transfer_files.wdl" as transfer_fastq_files
+import "../../tasks/utilities/data_import/task_create_terra_table.wdl" as terra_fastq_table
+
+workflow dorado_basecalling_workflow {
+  meta {
+    description: "GPU-accelerated workflow for basecalling Oxford Nanopore POD5 files, generating SAM outputs and supporting downstream demultiplexing and FASTQ output."
+  }
+
+  input {
+    Array[File] input_files
+    String dorado_model = "sup" 
+    String kit_name
+    String new_table_name
+    String fastq_upload_path
+    Boolean paired_end = false
+    Boolean assembly_data = false
+    String? file_ending
+    String terra_project
+    String terra_workspace
+    String fastq_file_name
+    Int cpu = 8                      
+    Int memory = 32                    
+    Int disk_size = 100               
+  }
+
+  scatter (file in input_files) {
+    call basecall_task.basecall as basecall_step {
+      input:
+        input_file = file,
+        dorado_model = dorado_model,
+        kit_name = kit_name,
+        cpu = cpu,
+        memory = memory,
+        disk_size = disk_size
+    }
+  }
+
+  call samtools_convert_task.samtools_convert {
+    input:
+      sam_files = flatten(basecall_step.sam_files)
+  }
+
+  call dorado_demux_task.dorado_demux {
+    input:
+      bam_files = samtools_convert.bam_files,
+      kit_name = kit_name,
+      fastq_file_name = fastq_file_name
+  }
+
+  call transfer_fastq_files.transfer_files as transfer_files {
+    input:
+      files_to_transfer = dorado_demux.fastq_files,
+      target_bucket = fastq_upload_path
+  }
+
+  if (defined(transfer_files.transferred_files)) {
+    call terra_fastq_table.create_terra_table as create_terra_table {
+      input:
+        new_table_name = new_table_name,
+        data_location_path = fastq_upload_path,
+        paired_end = paired_end,
+        assembly_data = assembly_data,
+        file_ending = file_ending,
+        terra_project = terra_project,
+        terra_workspace = terra_workspace
+    }
+  }
+
+  output {
+    Array[File] fastq_files = dorado_demux.fastq_files
```

Comment:
I'm concerned about the usability of this for labs. Wouldn't it be better to output this in such a way that it is accessible for follow-on workflows to use the fastq files (which generally take single inputs for read data, not arrays)?

Reply:
We could adjust the output to use flatten, allowing individual FASTQ files to be directly accessible. Just checking, do you know if Terra workflows handle flatten consistently for outputting individual files rather than arrays? Here’s the current output configuration:
output {
    Array[File] individual_fastq_files = flatten(dorado_demux.fastq_files)
    File? terra_table_tsv = create_terra_table.terra_table_to_upload
}

Reply:
You can disregard this comment. I overlooked the fact that you were also making these available via create_terra_table. Thanks @sage-wright for highlighting that to me. All good to leave them as an array.